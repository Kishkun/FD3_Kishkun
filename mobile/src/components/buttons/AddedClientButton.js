import React from 'react';
import myEvents from './../../events';

const AddedClientButton = (props) => {
    console.log('add button function');
    return (
        <button
            type='button'
            className='clients_btn add_btn'
            onClick={()=> myEvents.emit('onOpenAddedItem', this)}
        >Добавить клиента</button>
    )
};

export default AddedClientButton;