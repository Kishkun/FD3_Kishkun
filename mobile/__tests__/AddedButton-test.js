import React from 'react';
import AddedClientButton from '../src/components/buttons/AddedClientButton';
import myEvents from '../src/events';
import {shallow} from 'enzyme';

test('работа AddedClientButton', () => {
    const wrapper = shallow(<AddedClientButton/>);
    expect(wrapper).toMatchSnapshot();

    const buttonElem = wrapper.find('.add_btn');

    let eventTriggered = false;

    myEvents.on('onOpenAddedItem', function () {
        eventTriggered = true;
    });

    buttonElem.simulate('click');

    expect(eventTriggered).toBeTruthy();

});