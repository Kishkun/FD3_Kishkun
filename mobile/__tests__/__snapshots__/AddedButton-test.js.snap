// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа AddedClientButton 1`] = `
<button
  className="clients_btn add_btn"
  onClick={[Function]}
  type="button"
>
  Добавить клиента
</button>
`;
